backend/config/db_router.py

```
@@ -1,101 +1,211 @@
+        now = time.time()
+        return [r for r in self.replicas if self._revive_if_timeout(r, now)]
 
-        for replica in available:
-            connection = connections[replica]
-            if connection.connection is not None:
+    def _weighted_choice(self, available: list[str]) -> Optional[str]:
+        """Pick one replica using weighted random selection."""
+        if not available:
+            return None
+        total = sum(self._weights.get(r, 1) for r in available)
+        threshold = random.uniform(0, total)
```

Ruff (0.15.21)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)